util/test/checkChplInstall

```diff
@@ -75,11 +85,21 @@ else
     log_info "Found \$CHPL_HOME directory: ${CHPL_HOME}"
 fi
 
+CHPL_DIR_REMOVE=1
+
 # Create ~/.chpl directory, if it does not already exist
 CHPL_DIR=${CHPL_CHECK_INSTALL_DIR:-${HOME}/.chpl}
 if [ ! -d ${CHPL_DIR} ] ; then
     log_info "${CHPL_DIR} does not exist. Creating it."
     mkdir -p ${CHPL_DIR} || { log_error "Failed to create ${CHPL_DIR}." && exit 1 ; }
+else
+    log_info "${CHPL_DIR} already exists. Using it."
+    if find "${CHPL_DIR}" -mindepth 1 -print -quit | grep -q .; then
+        log_debug "${CHPL_DIR} is not empty. Not removing it after 'make check'"
+        CHPL_DIR_REMOVE=0
+    else
+        log_debug "${CHPL_DIR} is empty. Removing it after 'make check'"
```

Comment:
TThe argument here is that basically every Chapel user up to this point has this `.chpl` directory sitting in their home directory, because our `make check` script was sloppy about cleaning up in the past. This will remove those empty `~/.chpl` directories as well as clean up after itself in the future.

It's a little aggressive, but I don't expect it to cause anyone problems. However, I am okay with either approach. Maybe @bradcray would like to weigh in - he suggested the change.